tests/testthat/test-osmodel.R

```diff
@@ -0,0 +1,107 @@
+
+test_that("parametrize filled all the gaps", {
+    os_mod <- LogLogisticOs()
+    h_link <- HazardLink(
+        stan = StanModule(
+            functions = "bla",
+            data = "bla",
+            parameters = "bla",
+            transformed_parameters = "bla",
+            generated_quantities = "bla",
+            priors = list(),
+            inits = list()
+        ),
+        parameters = "bla",
+        contribution = "bla"
+    )
+    os_mod_f <- parametrize(os_mod, h_link)
+
+    expect_false(all(sapply(
+        c("link_arguments", "link_log_hazard_contribution", "link_arguments_as_par"),
+        grepl, os_mod_f@stan@functions
+    )))
+
+    expect_false(all(sapply(c("link_parameters"), grepl, os_mod_f@stan@parameters)))
+
+    expect_false(all(sapply(
+        c("link_log_surv", "link_log_lik"),
+        grepl, os_mod_f@stan@transformed_parameters
+    )))
+
+    expect_false(all(sapply(c("link_arguments_as_par"), grepl, os_mod_f@stan@generated_quantities)))
+})
+
+
+
+test_that("merge is working as expected", {
+    os_mod <- OsModel(stan = StanModule(
+        functions = " bla <link_arguments> bla <link_log_hazard_contribution> bla <link_arguments_as_par>",
+        data = "bla",
+        parameters = "bla <link_parameters>",
+        transformed_parameters = "bla <link_log_surv> bla <link_log_lik>",
+        generated_quantities = "bla <link_arguments_as_par>"
+    ))
+    h_link <- HazardLink(
+        stan = StanModule(
+            functions = "function",
+            data = "data",
+            parameters = "parameters",
+            transformed_parameters = "transformed_parameters",
+            generated_quantities = "generated_quantities",
+            priors = list(),
+            inits = list()
+        ),
+        parameters = "beta",
+        contribution = "contribution"
+    )
+    actual <- parametrize(os_mod, h_link)
+    expected <- OsModel(stan = StanModule(
+        functions = " bla real beta, bla betacontribution bla beta\n function",
+        data = "bla",
+        parameters = "bla parameters",
+        transformed_parameters = "bla beta, bla beta,",
+        priors = list(),
+        generated_quantities = "bla beta",
+        inits = list()
+    ))
+    expect_equal(actual, expected)
+})
+
+
+
+test_that("LogLogisticModule is working as expected", {
+    actual <- LogLogisticModule()
+    expected <- StanModule(
```

Comment:
I wonder if we should move towards using snapshots for testing, see https://testthat.r-lib.org/reference/expect_snapshot.html